Stellar-transaction.x

```diff
@@ -540,9 +540,23 @@ struct AddressWithNonce
     uint64 nonce;
 };
 
+enum AuthorizerType
+{
+    AUTHORIZER_SOURCE_ACCOUNT = 0,
+    AUTHORIZER_ADDRESS = 1,
+};
+
+union Authorizer switch (AuthorizerType type)
+{
+case AUTHORIZER_SOURCE_ACCOUNT:
+    void;
+case AUTHORIZER_ADDRESS:
+    AddressWithNonce address;
+};
+
 struct ContractAuth
 {
-    AddressWithNonce* addressWithNonce; // not present for invoker
+    Authorizer authorizer;
```

Comment:
Would it make sense to also move `signatureArgs` to `Authorizer`? This will auto-ensure the invariant of having no signatures for the invoker.

Reply:
That is less clear to me.

A reason to keep signatureArgs where it is, is that it is completely reasonable to say that some types of authorizers provide no signatures.

A motivation to move signatureArgs would be if we expect future types of authorizers to be able to pass a different structure as a signature. However, the downstream impact of this would be significant since everything downstream, including custom account contracts, assumes signatures are a vector.

So it sounds good, but not sure if the benefit is real. I think that change can be done and considered separately. This change is purely a naming change and is otherwise binary compatible with the existing XDR.

Reply:
Ok, I don't have a strong opinion on this, just a thought.

Reply:
I reckon we should do it. At minimum it better communicates the source account case. I'll open a second PR with it.

Reply:
Opened PR with the change at https://github.com/stellar/stellar-xdr/pull/89.